sled-agent/src/bootstrap/measurements.rs

```diff
@@ -0,0 +1,65 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Functions related to management of measurement corpus
+use camino::{Utf8Path, Utf8PathBuf};
+use sled_agent_config_reconciler::InternalDisksReceiver;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum MeasurementError {
+    #[error("Missing INSTALL dataset")]
+    MissingInstallSet,
+    #[error("io: {0}")]
+    Io(std::io::Error),
+    #[error("Missing boot disk")]
+    MissingBootDisk,
+}
+
+/// Access the measurements in the install directory
+pub async fn sled_new_measurement_paths(
+    receiver: &InternalDisksReceiver,
+) -> Result<Vec<Utf8PathBuf>, MeasurementError> {
+    let mut all = vec![];
+    let mut dirs = vec![];
+
+    let current = receiver.current();
+
+    dirs.push(
+        current
+            .boot_disk_install_dataset()
+            .ok_or(MeasurementError::MissingBootDisk)?
+            .join("measurements"),
+    );
+
+    if dirs.is_empty() {
+        return Err(MeasurementError::MissingInstallSet);
+    }
+
+    // We don't have an install dataset for our automated deployment
+    // testing. Instead, rely on the files getting copied/packaged.
+    let testing_corpus_path =
+        Utf8Path::new("/opt/oxide/sled-agent/pkg/testing-measurements");
+
+    if testing_corpus_path.is_dir() {
+        dirs.push(testing_corpus_path.into());
+    }
+
+    for dir in dirs {
+        match dir.read_dir_utf8() {
+            Ok(iter) => {
+                for entry in iter {
+                    let entry = entry.map_err(MeasurementError::Io)?;
```

Comment:
Should we also skip over this error as we do below?

Reply:
This is a different case, `read_dir_utf8` may return an error if the path doesn't exist. The iterator that returns is iterating over all files which I would expect to always work except if there are actual IO errors

Reply:
Makes sense.